core/src/main/java/org/apache/calcite/rel/metadata/RelMdCollation.java

```diff
@@ -240,7 +241,13 @@ private RelMdCollation() {}
    * {@link org.apache.calcite.rel.core.Calc}'s collation. */
   public static List<RelCollation> calc(RelMetadataQuery mq, RelNode input,
       RexProgram program) {
-    return program.getCollations(mq.collations(input));
```

Comment:
With this change, `RexProgram#getCollations` seems to be unused. Maybe we should tag it as deprecated for the next version, and eventually remove it in a subsequent version, to avoid leaving unnecessary code behind:
```
public class RexProgram {
...
  /**
   * Given a list of collations which hold for the input to this program,
   * returns a list of collations which hold for its output. The result is
   * mutable and sorted.
   * @deprecated Use {@link RelMdCollation#calc}
   */
  @Deprecated // to be removed before 2.0
  public List<RelCollation> getCollations(List<RelCollation> inputCollations) { ... }
```

Reply:
Yeah, I'm planning to do this.

Reply:
Ok, thanks @danny0405 !